File: Easy/0028_find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.go
package main

import (
	"fmt"
	"strings"
)

func strStr(haystack string, needle string) int {
	return strings.Index(haystack, needle)
}

type testCase struct {
	haystack string
	needle string
}

func main() {
	tests := []testCase{
		{haystack: "sadbutsad", needle: "sad"},
		{haystack: "leetcode", needle: "leeto"},
	}

	for _, tc := range tests {
		fmt.Printf("---\nhaystack: %s\nneedle: %s\nstrStr: %d\n", tc.haystack, tc.needle, strStr(tc.haystack, tc.needle))
	}
}

File: Easy/0028_find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
#!/bin/python3

from typing import List, Tuple

class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)

def main():
    tests : List[Tuple[str, str]] = [
        ("sadbutsad", "sad"),
        ("leetcode", "leeto"),
    ]

    for haystack, needle in tests:
        print("---\nhaystack: ", haystack, "\nneedle: ", needle, "\nstrStr: ", Solution().strStr(haystack, needle))

if __name__ == "__main__":
    main()
